src/crimson/osd/heartbeat.cc

```diff
@@ -135,11 +135,11 @@ seastar::future<Heartbeat::osds_t> Heartbeat::remove_down_peers()
         return seastar::make_ready_future<osd_id_t>(-1);
       }
     }, osds_t{},
-    [this](osds_t&& extras, osd_id_t extra) {
+    [](osds_t&& extras, osd_id_t extra) {
       if (extra >= 0) {
         extras.push_back(extra);
       }
-      return extras;
+      return std::move(extras);
```

Comment:
One minor note here: I was curious about this warning. I wasn't able to find a difference in the actual code produced with or without the move().
(Anyway - adding the move() removes the warning)

Reply:
my understanding is the C++17 does not enforce the copy elision in this case. but since `extra` is already moved away, the compiler is aware of this, so it applies the optimization anyway with warning messages. please note the warning was, well, just a warning. it does not imply that the optimization is disabled.